plugins/cairo-lang-macro/src/types/token.rs

```diff
@@ -0,0 +1,149 @@
+use std::fmt::Display;
+
+use cairo_lang_diagnostics::DiagnosticsBuilder;
+use cairo_lang_filesystem::{
+    ids::{FileKind, FileLongId, VirtualFile},
+    span::TextSpan,
+};
+use cairo_lang_parser::{parser::Parser, utils::SimpleParserDatabase};
+use cairo_lang_syntax::node::TypedSyntaxNode;
+use cairo_lang_syntax::node::{db::SyntaxGroup, SyntaxNode};
+use cairo_lang_utils::Intern;
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+
+/// Representation of most atomic struct that is passed within host<->macro communication.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct Token {
+    span: TextSpan,
+    content: String,
+}
+
+impl Token {
+    pub fn new(content: String, span: TextSpan) -> Self {
+        Self { content, span }
+    }
+
+    pub fn from_syntax_node(db: &dyn SyntaxGroup, node: SyntaxNode) -> Self {
```

Comment:
This package should not depend on Cairo compiler at all. Instead of accepting SyntaxNode and building stream for the user, it should simply define the structs that user needs to build himself.